packages/web3torrent/src/hooks/use-budget.ts

```diff
@@ -16,6 +16,7 @@ export function useBudget({ready}: {ready: boolean}) {
       setLoading(false);
     };
     if (ready) getAndSetBudget();
+    return paymentChannelClient.channelClient.onBudgetUpdated(getAndSetBudget);
```

Comment:
I did this so that components that use this hook will get re-rendered when a `'BudgetUpdated'` event fires. I had to make the `channelClient` a public property to do this.

Reply:
@kerzhner be good to get your thoughts.

Reply:
I think this solves the problem of the budget not always displaying: you might have a budget but your react components don't know about it yet. #2042